### PR TITLE
build(deps): bump @sentry/react-native from 5.9.1 to 5.14.1

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-435] 

### Description

Bumps sentry to latest - triggering beta to see if sourcemaps work correctly

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-  bump @sentry/react-native from 5.9.1 to 5.14.1 - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-435]: https://artsyproduct.atlassian.net/browse/PHIRE-435?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MrSltun — APPROVED** [review]

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (bump @sentry/react-native from 5.9.1 to 5.14.1 - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 915fad54d51c50a3779778be22cd475ba1145d94
</p>

**Comment by gkartalis** [comment]
#DO NOT MERGE - till we see that the sourcemaps work correctly

**Comment by gkartalis** [comment]
tested on iOS ~ https://artsynet.sentry.io/issues/4459370240/events/e6bcfb7e358d45c2a5975a604dfd2d01/

still needs to be tested on Android (beta version: `android-8.27.0-1674645525`)

File: ios/Podfile.lock [hunk 1]
```diff
@@ -698,9 +698,11 @@ PODS:
   - RNScreens (3.26.0):
     - React-Core
     - React-RCTImage
-  - RNSentry (5.9.1):
```
Comment (brainbicycle) [inline ios/Podfile.lock:701]: We have some native code using sentry SDK, just double checking no updates needed there? If it compiles and runs it should be okay `ARSentryReporter.m` and `ARNetworkErrorManager.m`

Comment (gkartalis) [reply ios/Podfile.lock:701]: it does compile and the beta seems to be working fine, is there something that I need to look into further?

Comment (brainbicycle) [reply ios/Podfile.lock:701]: no should be alright then

**Pull request merged**